packages/components/nodes/moderation/SimplePromptModeration/SimplePromptModerationRunner.ts

```diff
@@ -1,23 +1,38 @@
 import { Moderation } from '../Moderation'
+import { BaseChatModel } from 'langchain/chat_models/base'
 
 export class SimplePromptModerationRunner implements Moderation {
     private readonly denyList: string = ''
     private readonly moderationErrorMessage: string = ''
+    private readonly model: BaseChatModel
 
-    constructor(denyList: string, moderationErrorMessage: string) {
+    constructor(denyList: string, moderationErrorMessage: string, model?: BaseChatModel) {
         this.denyList = denyList
         if (denyList.indexOf('\n') === -1) {
             this.denyList += '\n'
         }
         this.moderationErrorMessage = moderationErrorMessage
+        if (model) this.model = model
     }
 
     async checkForViolations(input: string): Promise<string> {
-        this.denyList.split('\n').forEach((denyListItem) => {
-            if (denyListItem && denyListItem !== '' && input.toLowerCase().includes(denyListItem.toLowerCase())) {
-                throw Error(this.moderationErrorMessage)
+        if (this.model) {
+            const denyArray = this.denyList.split('\n')
+            for (const denyStr of denyArray) {
+                const res = await this.model.invoke(
```

Comment:
denyString can be empty string, if there are trailing \n

Reply:
updated check